### PR TITLE
Improve & simplify readme & Add contributing file

### PR DESCRIPTION
**Pull request by maxnorm**
Restructures the project documentation to better separate user-facing information from contributor-focused details.

### README.md
- Added table of content
- Reduced from 8 detailed sections to concise bullet points with references to CONTRIBUTING.md
- Brief overview with reference to CONTRIBUTING.md for complete details
- Essential commands only (build, test, format) with reference to CONTRIBUTING.md

### CONTRIBUTING.md
- Complete detailed explanations, examples, and philosophical context
- All 7 detailed design principles with full explanations and examples like in the README before.
- Detailed usage examples, test options, and help commands

Related issue: 
- https://github.com/Perfect-Abstractions/Compose/issues/39
- https://github.com/Perfect-Abstractions/Compose/issues/61

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+   ```bash
+   forge test
+   ```
+
+## Development Workflow
+
+Once you are assigned to an issue:
+1. Fork the repository
+2. Create a new branch for your work
+3. Implement your changes
```
Comment (mudgen) [inline CONTRIBUTING.md:74]: 4. Format your code (`forge fmt`)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+forge test --gas-report
+```
+
+### Test Writing Guidelines
+- Write comprehensive tests for all new functionality
+- Test edge cases and error conditions
+- Use descriptive test names
+- Follow the existing test patterns in the codebase
+- Include gas optimization tests where relevant
+
```
Comment (mudgen) [inline CONTRIBUTING.md:267]: Add a note here that if you don't want to write tests for new functionality then we can find someone to work with you to write the tests. Only make a pull request with full test coverage of your code.

**Comment by mudgen** [comment]
@maxnorm This is a pretty great change. I like it a lot. I made a couple comments. Please address my comments and let's get this merged. :rocket:

**Comment by mudgen** [comment]
Should also mention that we are using Github's [dicussions](https://github.com/Perfect-Abstractions/Compose/discussions) to discuss things, ideas etc.  Also questions can be asked in discussions.  It is a forum for the project.

Issues are for actions to be taken.   Discussions are for general communication and discussing things, talking about possibilities and ideas etc.

**Comment by maxnorm** [comment]
@mudgen It's now adjust based on your reviews.

**Comment by mudgen** [comment]
Thanks!
